components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -718,13 +718,56 @@ private boolean isDomainNameEmbeddedInAttributeValue(String filterOperation, Str
             String[] userStoreDomainNames = getDomainNames();
             ArrayList<String> filteredUsernames = new ArrayList<>();
             for (String userStoreDomainName : userStoreDomainNames) {
-                filteredUsernames.addAll(Arrays
-                        .asList(getFilteredUsernames(node, offset, limit, sortBy, sortOrder, userStoreDomainName)));
+                String[] userNames = getFilteredUsernames(node, offset, limit, sortBy, sortOrder, userStoreDomainName);
+
+                // Calculating new offset and limit parameters.
+                int[] newParams = calculateOffsetAndLimit(node, offset, limit, sortBy, sortOrder, domainName,
```

Comment:
can userNames be null? if so we need to handle it properly

Reply:
Fixed: 034f63e984cc91211fb945bc398451d985fa7f51